code/clue.js

```diff
@@ -1,59 +1,464 @@
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Mystery storage
+
+const mystery = {
+  killer: undefined,
+  favouriteWeapon: undefined,
+  pickedWeapon: undefined,
+  weapon: undefined,
+  favouriteRoom: undefined,
+  pickedRoom: undefined,
+  room: undefined,
+};
+
+// This function will change favourite weapons of each person
+
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((item) => {
+    item.favouriteWeapon = randomSelector(weapons).name;
+  });
+  if (mystery.killer !== undefined) {
+    mystery.favouriteWeapon = mystery.killer.favouriteWeapon;
+    killerFavouriteWeapon.innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+  }
+};
+
+// Actions related to Killer deck
+
+const killerAnimation = () => {
+  killerLoader.style.display = "block";
+  setTimeout(pickKiller, 2000);
+  revealedMystery.innerHTML = '';
+  killerCard.style.background = "";
+  killerName.innerHTML = '';
+  killerAge.innerHTML = '';
+  killerFavouriteWeapon.innerHTML = '';
+  killerImage.style.display = "none";
+  killerImage.src = '';
+};
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  mystery.favouriteWeapon = mystery.killer.favouriteWeapon;
+  killerLoader.style.display = "none";
+  killerCard.style.background = mystery.killer.color;
+  killerName.innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerAge.innerHTML = `Age: ${mystery.killer.age}`;
+  killerFavouriteWeapon.innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+  killerImage.style.display = "inline-block";
+  killerImage.src = mystery.killer.image;
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+killerCard.onclick = killerAnimation;
+
+// Actions related to Weapon deck
+
+const favouriteWeaponAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else {
+    weaponLoader.style.display = "block";
+    weaponName.innerHTML = '';
+    weaponWeight.innerHTML = '';
+    weaponImage.src = '';
+    weaponImage.style.display = "none";
+    weaponFavouriteRoom.innerHTML = '';
+    revealedMystery.innerHTML = '';
+    setTimeout(function () {
```

Comment:
oh no! A `function` 🚫! Arrow functions are the way => {😬}

Reply:
@sofiavazs I found a description of the setTimeOut method on w3schools (https://www.w3schools.com/jsref/met_win_settimeout.asp) and I did not dare to change it 😬. But now, I did it, thanks to your comment! 💐0